packages/mui-base/src/unstable_useModal/useModal.types.ts

```diff
@@ -0,0 +1,113 @@
+import { PortalProps } from '../Portal';
+
+export interface UseModalRootSlotOwnProps {
+  role: React.AriaRole;
+  onKeyDown: React.KeyboardEventHandler;
+  ref: React.RefCallback<Element> | null;
+}
+
+export interface UseModalBackdropSlotOwnProps {
+  'aria-hidden': React.AriaAttributes['aria-hidden'];
+  onClick: React.MouseEventHandler;
+  open?: boolean;
+}
+
+export type UseModalBackdropSlotProps<TOther = {}> = TOther & UseModalBackdropSlotOwnProps;
+
+export type UseModalRootSlotProps<TOther = {}> = TOther & UseModalRootSlotOwnProps;
+
+export type UseModalParameters = {
+  'aria-hidden'?: React.AriaAttributes['aria-hidden'];
```

Comment:
Why would it be unusual to have it? If we don't have it, we would need to duplicate this logic three times and teach people how to process the prop before sending it to the hook under some other name. It's clear what it is. For e.g. react aria has this often, for e.g. https://react-spectrum.adobe.com/react-aria/useSelect.html.

Reply:
> Having a verbatim aria-hidden may suggest it's just forwarded as an element prop.

I understand but why is this bad thing? This is what I am trying to understand. The hooks are intended to be used in components, so why is it bad thing to just forward some prop?

Reply:
My point is that it may suggest it's just forwarded, but there is some logic associated with it, so it won't always end up being forwarded.
Plus, it's easier to specify `hidden` in the hook's parameters than `'aria-hidden'`.

Reply:
I feel like it's straight forward, let's see if we will have complains about it. The hook is marked as `unstable_` we can change if we see opportunities to improve.